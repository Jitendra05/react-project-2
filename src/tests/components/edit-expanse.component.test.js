import React from 'react';
import {shallow} from 'enzyme';
import {EditExpanseComponent} from '../../components/edit-expanse.component';
import {expanses} from '../fixtures/expanses-data';
import {
  startEditExpanse,
} from '../../actions/expanse-action';
let startEditExpanseSpy, okPopupSpy, historySpy, wrapper;

beforeEach(()=>{
  okPopupSpy = jest.fn();
  startEditExpanseSpy =  jest.fn();
  historySpy = {push:jest.fn()};
  wrapper = shallow(
    <EditExpanseComponent 
        startEditExpanse={startEditExpanseSpy} 
        history={historySpy} 
        expanse={expanses[2]}
        okPopup={okPopupSpy}
    />);
});

test('Should render EditExpanseComponent correctly',()=>{
  expect(wrapper).toMatchSnapshot();
});


test('Should set EditExpanseComponent state correctly onSubmit',()=>{
  wrapper.find('ExpanseForm').prop('onSubmit')(expanses[1]);
  expect(wrapper.state('isSuccess')).toBeFalsy();
  expect(wrapper.state('isWarning')).toBeTruthy();
  expect(wrapper.state('id')).toEqual(expanses[2].id);
  expect(wrapper.state('updatedExpanse')).toEqual(expanses[1]);
});



// test('Should handle editExpanse on okPopup click',()=>{
//   wrapper.find('#warningModal').prop('okPopup')();
//   expect(okPopupSpy).toHaveBeenCalled();
// });
