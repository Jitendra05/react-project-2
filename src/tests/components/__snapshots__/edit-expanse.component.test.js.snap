// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditExpanseComponent correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="container"
    >
      <h2
        className="page-header__title"
      >
        Edit Expense
      </h2>
    </div>
  </div>
  <div
    className="container"
  >
    <ExpanseForm
      expanse={
        Object {
          "amount": 1200,
          "createdAt": 4000,
          "description": "water bill",
          "id": "3",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <WarningModal
      closePopup={[Function]}
      id="warningModal"
      isVisible={false}
      message="Are you sure? You want to update the expanse."
      okPopup={[Function]}
      title="Warning"
    />
    <SuccessModal
      id="SuccessModal"
      isVisible={false}
      message="Expanse updated sucessfuly."
      title="Success"
    />
  </div>
</div>
`;
